pkg/machine/wsl/machine.go

```diff
@@ -0,0 +1,1119 @@
+		return errors.Wrapf(err, "could not create '%s' authorized keys on guest OS", v.RemoteUsername)
+	}
+
+	return nil
+}
+
+func configureSystem(v *MachineVM, dist string) error {
+	user := v.RemoteUsername
+	if err := runCmdPassThrough("wsl", "-d", dist, "sh", "-c", fmt.Sprintf(appendPort, v.Port, v.Port)); err != nil {
+		return errors.Wrap(err, "could not configure SSH port for guest OS")
+	}
+
+	if err := pipeCmdPassThrough("wsl", withUser(configServices, user), "-d", dist, "sh"); err != nil {
+		return errors.Wrap(err, "could not configure systemd settings for guest OS")
+	}
+
+	if err := pipeCmdPassThrough("wsl", sudoers, "-d", dist, "sh", "-c", "cat >> /etc/sudoers"); err != nil {
+		return errors.Wrap(err, "could not add wheel to sudoers")
+	}
+
+	if err := pipeCmdPassThrough("wsl", overrideSysusers, "-d", dist, "sh", "-c",
+		"cat > /etc/systemd/system/systemd-sysusers.service.d/override.conf"); err != nil {
+		return errors.Wrap(err, "could not generate systemd-sysusers override for guest OS")
+	}
+
+	lingerCmd := withUser("cat > /home/[USER]/.config/systemd/[USER]/linger-example.service", user)
+	if err := pipeCmdPassThrough("wsl", lingerService, "-d", dist, "sh", "-c", lingerCmd); err != nil {
+		return errors.Wrap(err, "could not generate linger service for guest OS")
+	}
+
+	if err := pipeCmdPassThrough("wsl", withUser(lingerSetup, user), "-d", dist, "sh"); err != nil {
+		return errors.Wrap(err, "could not configure systemd settomgs for guest OS")
+	}
+
+	if err := pipeCmdPassThrough("wsl", containersConf, "-d", dist, "sh", "-c", "cat > /etc/containers/containers.conf"); err != nil {
+		return errors.Wrap(err, "could not create containers.conf for guest OS")
+	}
+
+	return nil
+}
+
+func installScripts(dist string) error {
+	if err := pipeCmdPassThrough("wsl", enterns, "-d", dist, "sh", "-c",
+		"cat > /usr/local/bin/enterns; chmod 755 /usr/local/bin/enterns"); err != nil {
+		return errors.Wrap(err, "could not create enterns script for guest OS")
+	}
+
+	if err := pipeCmdPassThrough("wsl", profile, "-d", dist, "sh", "-c",
+		"cat > /etc/profile.d/enterns.sh"); err != nil {
+		return errors.Wrap(err, "could not create motd profile script for guest OS")
+	}
+
+	if err := pipeCmdPassThrough("wsl", wslmotd, "-d", dist, "sh", "-c", "cat > /etc/wslmotd"); err != nil {
+		return errors.Wrap(err, "could not create a WSL MOTD for guest OS")
+	}
+
+	if err := pipeCmdPassThrough("wsl", bootstrap, "-d", dist, "sh", "-c",
+		"cat > /root/bootstrap; chmod 755 /root/bootstrap"); err != nil {
+		return errors.Wrap(err, "could not create bootstrap script for guest OS")
+	}
+
+	return nil
+}
+
+func checkAndInstallWSL(opts machine.InitOptions) (bool, error) {
+	if isWSLInstalled() {
+		return true, nil
+	}
+
+	admin := hasAdminRights()
+
+	if !isWSLFeatureEnabled() {
+		return false, attemptFeatureInstall(opts, admin)
+	}
+
+	skip := false
+	if !opts.ReExec && !admin {
+		fmt.Println("Launching WSL Kernel Install...")
```

Comment:
We are already using `logrus` in this package. Should we use for this as well.

Reply:
Similar explanation as above, this is part of an interactive installation session.